sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupAsyncClient.java

```diff
@@ -156,19 +178,19 @@ public String getVaultUrl() {
         };
     }
 
-    private Mono<PollResponse<FullBackupOperation>> processBackupOperationResponse(Response<FullBackupOperation> response) {
+    private Mono<PollResponse<KeyVaultBackupOperation>> processBackupOperationResponse(Response<KeyVaultBackupOperation> response) {
         String operationStatus = response.getValue().getStatus().toLowerCase();
 
         return Mono.just(new PollResponse<>(toLongRunningOperationStatus(operationStatus), response.getValue()));
     }
 
     private LongRunningOperationStatus toLongRunningOperationStatus(String operationStatus) {
         switch (operationStatus) {
```

Comment:
Would it make more sense to lower-case these and compare case-insensitively? I wouldn't trust the service implementation so much in this regard.

Reply:
Sounds good. I will add that small change.

Reply:
nit: To guard against possible string based breaking changes in the API, I decided to base this status on the state of both the `error` and `endTime` values.